tests/deploy/e2e/test_initialize.py

```diff
@@ -343,3 +344,105 @@ async def get_last_deployed() -> dict[int, datetime.datetime]:
             last_deployed=last_deployed_after[6],
         ),
     }
+
+
+@pytest.mark.parametrize("no_agent", (True,))
+@pytest.mark.parametrize("reset_state", (True, False))
+async def test_scheduler_initialize_multiple_versions(
+    agent_factory,
+    resource_container,
+    clienthelper,
+    server,
+    client,
+    environment,
+    reset_state: bool,
+) -> None:
+    """
+    Verify processing of multiple versions when the scheduler initializes from an empty state (first start for an
+    environment or reset_state). Concretely, verify that resources from the first version that are orphaned by the
+    second, are in fact marked as orphans.
+
+    :param reset_state: If True, achieve empty state through state reset. Otherwise achieve it via an empty environment.
+    """
+    if reset_state:
+        await client.set_setting(environment, data.RESET_DEPLOY_PROGRESS_ON_START, True)
+
+    async def start_agent() -> Agent:
+        agentmanager = server.get_slice(inmanta.server.SLICE_AGENT_MANAGER)
+
+        agent: Agent = await agent_factory(uuid.UUID(environment))
+        await utils.retry_limited(lambda: len(agentmanager.sessions) == 1, 10)
+        return agent
+
+    rid1: str = "test::Resource[agent1,key=key1]"
+    rid2: str = "test::Resource[agent1,key=key2]"
+    agent: Agent | None = None
+
+    def resources(version: int, *, r2: bool = True) -> list[dict[str, object]]:
+        def res(n: int) -> None:
+            return {
+                "key": f"key{n}",
+                "value": "val{i}",
```

Comment:
Is this intentional ?
```suggestion
                "value": f"val{n}",
```